src/Inciter/AMR/util.cpp

```diff
@@ -0,0 +1,108 @@
+#include "util.h"
+
+namespace AMR {
+    namespace util {
+
+        /**
+         * @brief Split a string based on a given delimiter
+         *
+         * @param s The string to split
+         * @param delim The delimiter to use
+         * @param elems The vector which will be filled
+         */
+        void split(const std::string &s, char delim,
+                std::vector<std::string> &elems)
+        {
+            std::stringstream ss;
+            ss.str(s);
+            std::string item;
+            while (std::getline(ss, item, delim)) {
+                elems.push_back(item);
+            }
+        }
+
+        /**
+         * @brief Build a vector of split strings based on a delmiter
+         *
+         * @param s The string to split
+         * @param delim The delimiter to use
+         *
+         * @return A vector split by delim. Does not handle empty tokens
+         */
+        std::vector<std::string> split(const std::string &s, char delim) {
+            std::vector<std::string> elems;
+            split(s, delim, elems);
+            return elems;
+        }
+
+
+        /**
+         * @brief function to find the mid point betwen two points (nodes)
+         *
+         * @param edge_node_A First node/point
+         * @param edge_node_B Second node/point
+         *
+         * @return  The mid point between the two nodes
+         */
+        coordinate_t find_mid_point(
+                coordinate_t edge_node_A,
+                coordinate_t edge_node_B
+                )
+        {
+            coordinate_t mid_point;
+
+            for(size_t i = 0; i < DIMENSION; i++)
+            {
+                mid_point[i] = (edge_node_A[i]+edge_node_B[i])/2.0;
+            }
+
+            return mid_point;
+        }
```

Comment:
Ah, OK. Somewhere has to track the coordinates of where the new AMR nodes are added, and the AMR lib isn't an unreasonable place to do this [assume geom tracking is enabled]. If that is true, then this kind of functionality doesn't fundamentally not make sense